sklearn/_loss/_loss.pxd

```diff
@@ -1,20 +1,15 @@
 # cython: language_level=3
 
-cimport numpy as cnp
-
-cnp.import_array()
-
-
 # Fused types for y_true, y_pred, raw_prediction
 ctypedef fused Y_DTYPE_C:
-    cnp.npy_float64
-    cnp.npy_float32
+    double
+    float
 
```

Comment:
Should we wait until this is fixed upstream in Cython?

Reply:
For me, I prefer to use `double` and `float` because it can take a while for Cython to backport a fix to a release. (It took ~3 years to backport the const + fused type fix).

In this case, I even prefer `double` and `float` because `_loss.pyx.tp` only uses `double` to represent `float64`.

**Aside**: Here is where NumPy defines `float32` and `float64` to be `float` and `double`: https://github.com/numpy/numpy/blob/c4c0bbd36e597f24c481ffd736527cc48309e481/numpy/__init__.cython-30.pxd#L306-L307

Reply:
I agree with @thomasjpfan.

I think that we might want to clarify `dtype`'s and types' conventions and usages in scikit-learn. This was initiated by https://github.com/scikit-learn/scikit-learn/pull/24153 but it got stalled.

What do you think?

Reply:
I opened #25572.

Reply:
As I said on the linked issue, I prefer type names with explicit precision levels (both for integers and floating point values).